mgmt/rest/v2/api.go

```diff
@@ -56,25 +56,277 @@ func New(wg *sync.WaitGroup, killChan chan struct{}, protocol string) *apiV2 {
+		//
+		// Consumes:
+		// application/json
+		// application/x-protobuf
+		// text/plain
+		//
+		// Produces:
+		// application/json
+		// application/x-protobuf
+		// text/plain
+		//
+		// Schemes: http, https
+		//
+		// Responses:
+		// 204: PluginResponse
+		// 400: ErrorResponse
+		// 404: ErrorResponse
+		// 409: ErrorResponse
+		// 500: ErrorResponse
 		api.Route{Method: "DELETE", Path: prefix + "/plugins/:type/:name/:version", Handle: s.unloadPlugin},
-
+		// swagger:route GET /plugins/{ptype}/{pname}/{pversion}/config getPluginConfigItem
+		//
+		// lists the config of a giving plugin. The allowed plugin types are collector, processor, and publisher.
+		// Any other type results in error.
+		//
+		// Consumes:
+		// application/json
+		// application/x-protobuf
+		//
+		// Produces:
+		// application/json
+		// application/x-protobuf
+		//
+		// Schemes: http, https
+		//
+		// Responses:
+		// 200: PluginConfigResponse
+		// 400: ErrorResponse
 		api.Route{Method: "GET", Path: prefix + "/plugins/:type/:name/:version/config", Handle: s.getPluginConfigItem},
+		// swagger:route PUT /plugins/{ptype}/{pname}/{pversion}/config setPluginConfigItem
+		//
+		// updates the config of a giving plugin. A wrong plugin type or non-numeric plugin version results in error.
+		//
+		// Consumes:
+		// application/json
+		// application/x-protobuf
+		//
+		// Produces:
+		// application/json
+		// application/x-protobuf
+		//
+		// Schemes: http, https
+		//
+		// Responses:
+		// 200: PluginConfigResponse
+		// 400: ErrorResponse
 		api.Route{Method: "PUT", Path: prefix + "/plugins/:type/:name/:version/config", Handle: s.setPluginConfigItem},
+		// swagger:route DELETE /plugins/{ptype}/{pname}/{pversion}/config deletePluginConfigItem
+		//
+		// deletes the config of a giving plugin. Note that that to be removed config items are a slice of config keys.
+		// At lease one config key is required for this operation. An error occurs for any bad request.
+		//
+		// Consumes:
+		// application/json
+		// application/x-protobuf
+		//
+		// Produces:
+		// application/json
+		// application/x-protobuf
+		//
+		// Schemes: http, https
+		//
+		// Responses:
+		// 200: PluginConfigResponse
+		// 400: ErrorResponse
 		api.Route{Method: "DELETE", Path: prefix + "/plugins/:type/:name/:version/config", Handle: s.deletePluginConfigItem},
-
-		// metric routes
+		// swagger:route GET /metrics getMetrics
+		//
+		// lists a list of loaded metric types. An empty list returns if there is no loaded metrics. Any bad request results in error.
+		//
+		// Consumes:
+		// application/json
+		// application/x-protobuf
+		//
+		// Produces:
+		// application/json
+		// application/x-protobuf
+		//
+		// Schemes: http, https
+		//
+		// Responses:
+		// 200: MetricsResponse
+		// 404: ErrorResponse
+		// 500: ErrorResponse
 		api.Route{Method: "GET", Path: prefix + "/metrics", Handle: s.getMetrics},
```

Comment:
@andrzej-k, it's supported in v2 as well. it's rolled into one API instead of 2APIs.